src/hooks/useTxDetails/useTxDetails.ts

```diff
@@ -106,30 +106,28 @@ export const useTxDetails = (txId: string): TxDetails => {
   const tx = useAppSelector((state: ReduxState) => selectTxById(state, txId))
   const assets = useAppSelector(selectAssets)
 
-  const transfers = useMemo(() => {
-    if (!tx) return []
-    return getTransfers(tx, assets)
-  }, [tx, assets])
+  // This should never happen, but if it does we should not continue
+  if (!tx) throw Error('Transaction not found')
```

Comment:
Is it going to be caught by the error boundary and potentially produce a holistic crash?

Reply:
Actually disregard me, this was already present and never produced any crash AFAIK... or maybe it did and that infamous first load of ETH page is related to it? Would probably just ditch it and go with the optional assumption everywhere in this hook while at it 🙏🏽

Reply:
Worth looking into and updating to be more graceful. This would have been changed in: https://github.com/shapeshift/web/pull/7366.